### DISCUSSION TITLE
Browser mvvm error

### DISCUSSION THREAD

**Discussion by bbhxwl** [AvaloniaUI/Avalonia#15946] [Q&A]
### Describe the bug

The browser's display of text is also abnormal, and it cannot be automatically updated by MVVM, but Windows can function normally.

```
using ReactiveUI;

namespace AvaloniaApplication2.ViewModels;

public class MainViewModel : ViewModelBase
{
    private string _greeting =  "Welcome to Avalonia!";
    public string Greeting
    {
        get => _greeting;
        set => this.RaiseAndSetIfChanged(ref _greeting, value);
    }
}
```
```
using Avalonia.Controls;
using Avalonia.Interactivity;
using AvaloniaApplication2.ViewModels;

namespace AvaloniaApplication2.Views;

public partial class MainView : UserControl
{
    MainViewModel viewModel;
    public MainView()
    {
        InitializeComponent();
        viewModel=new MainViewModel();
        DataContext = viewModel;
    }

    private void Button_OnClick(object? sender, RoutedEventArgs e)
    {
        viewModel.Greeting = "11111111111";
    }
}
```
![image](https://github.com/AvaloniaUI/Avalonia/assets/26525599/9eda2f90-43bc-4050-ba9c-22d0fee31765)
![image](https://github.com/AvaloniaUI/Avalonia/assets/26525599/0b6279e4-d9e9-4658-9e75-51fce466aeee)
![image](https://github.com/AvaloniaUI/Avalonia/assets/26525599/1c28103e-826b-4784-9910-12ee232638b4)

### To Reproduce

[AvaloniaApplication2.zip](https://github.com/user-attachments/files/15688771/AvaloniaApplication2.zip)

### Avalonia version

11.0.10

**Comment by rabbitism** [comment]
![image](https://github.com/AvaloniaUI/Avalonia/assets/14807942/86a7d28a-7204-44b2-b5e9-1f86744c5785)
Because in browser app it goes to this line and your data context is re-assigned after constructor.

**Comment by bbhxwl** [comment]
> ![image](https://private-user-images.githubusercontent.com/14807942/337503565-86a7d28a-7204-44b2-b5e9-1f86744c5785.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc3MzA2NzgsIm5iZiI6MTcxNzczMDM3OCwicGF0aCI6Ii8xNDgwNzk0Mi8zMzc1MDM1NjUtODZhN2QyOGEtNzIwNC00NGIyLWI1ZTktMWY4Njc0NGM1Nzg1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA3VDAzMTkzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE1ODhiMzhiZmNmNDg3ZWM0YWJmYWUxYWNiYjVkN2MyZTVmZDFlODAxNWE1YTlkNmI2NTExODEzODE1MzZjMDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NvNDh7GciSyTq7k8AQFItILF4Im4S-Z-Ko0iLMe8dRU) Because in browser app it goes to this line and your data context is re-assigned after constructor.

Thank you, I am a beginner. Thank you very much. I have used xamarin before. But now the dilemma is whether to learn Maui or this.

**Reply by thevortexcloud** [reply]
Based on the general reaction I hear, Avalonia is the better choice. MAUI is apparently still fairly broken. Also MAUI does not support Linux, which makes it unusable for many, many people.

**Comment by rabbitism** [comment]
You can join QQ group 397510870 if you want to learn more about Avalonia.

**Comment by bbhxwl** [comment]
> You can join QQ group 397510870 if you want to learn more about Avalonia.

+了。没同意呢还
